<?xml version="1.0"?>
<launch>

    <!-- declare arg to be passed in -->
    <arg name="role" default="BIG_MAC" />
    
    <!-- Hardware agnostic code -->
    <node name="ball_estimator" pkg="playground" type="ball_estimator_node.py" />
    <node name="robot_estimator" pkg="playground" type="robot_estimator_node.py">
        <remap from="robot_estimator" to="ally1_estimator" />
        <remap from="vision_position" to="vision_ally1_position" />
    </node>
    <node name="robot_estimator" pkg="playground" type="robot_estimator_node.py">
        <remap from="robot_estimator" to="ally2_estimator" />
        <remap from="vision_position" to="vision_ally2_position" />
    </node>
    <node name="robot_estimator" pkg="playground" type="robot_estimator_node.py">
        <remap from="robot_estimator" to="opponent1_estimator" />
        <remap from="vision_position" to="vision_opponent1_position" />
    </node>
    <node name="robot_estimator" pkg="playground" type="robot_estimator_node.py">
        <remap from="robot_estimator" to="opponent2_estimator" />
        <remap from="vision_position" to="vision_opponent2_position" />
    </node>
    <node name="guidedog" pkg="playground" type="guidedog_node.py" output="screen" />
    <node name="controller" pkg="playground" type="controller_node.py">
        <remap from="desired_position" to="desired_position_safe" />
    </node>

    <!-- Hardware Dependent -->
    <node name="motion" pkg="playground" type="motion_node.py" />
</launch>
